src/protocols/user_directed_commissioning/UserDirectedCommissioningClient.cpp

```diff
@@ -44,7 +44,7 @@ CHIP_ERROR UserDirectedCommissioningClient::SendUDCMessage(TransportMgrBase * tr
     // send UDC message 5 times per spec (no ACK on this message)
     for (unsigned int i = 0; i < 5; i++)
     {
-        err = transportMgr->SendMessage(peerAddress, std::move(payload));
+        err = transportMgr->SendMessage(peerAddress, payload.CloneData());
```

Comment:
Would `Retain()` be enough? Do we need a deep copy?

Reply:
My understanding is that `SendMessage` takes over the payload, so it can be retained, but there is no guarantee that the payload is not changed...